### PR TITLE
bug fix: provider con not get source micro service name

### PR DESCRIPTION
**Pull request by lizihuai**

### CODE REVIEW

**Comment by tianxiaoliang** [comment]
Can one of the admins verify this patch?

File: core/common/common.go [hunk 1]
```diff
@@ -220,25 +226,33 @@ func GetXCSEHeader(k string, r *http.Request) string {
 }
 
 // SetXCSEHeader  set value into x-cse-header
-func SetXCSEHeader(k, v string, r *http.Request) {
-	if v == "" || k == "" || r == nil {
-		openlogging.GetLogger().Warn("set x_cse_header into req failed ,because one of key,value and request is empty(nil) or all empty(nil)")
+func SetXCSEHeader(vm map[string]string, r *http.Request) {
+	if len(vm) <= 0 || vm == nil || r == nil {
+		openlogging.GetLogger().Warn("set x-cse-header into req failed ,because one of key,value and request is empty(nil) or all empty(nil)")
```
Comment (tianxiaoliang) [inline core/common/common.go]: debug log

File: core/common/common.go [hunk 2]
```diff
@@ -201,12 +207,12 @@ func FromContext(ctx context.Context) map[string]string {
 // GetXCSEHeader  get x-cse-header from req.header
 func GetXCSEHeader(k string, r *http.Request) string {
 	if r == nil || k == "" {
-		openlogging.GetLogger().Error("get x_cse_header failed , request is nil or key is empty, please check its")
+		openlogging.GetLogger().Error("get x-cse-header failed , request is nil or key is empty, please check its")
```
Comment (tianxiaoliang) [inline core/common/common.go]: debug log

File: core/common/common.go [hunk 3]
```diff
@@ -201,12 +207,12 @@ func FromContext(ctx context.Context) map[string]string {
 // GetXCSEHeader  get x-cse-header from req.header
 func GetXCSEHeader(k string, r *http.Request) string {
```
Comment (tianxiaoliang) [inline core/common/common.go]: GetXCSEHeader -- GetXCSEContext

File: core/common/common.go [hunk 4]
```diff
@@ -201,12 +207,12 @@ func FromContext(ctx context.Context) map[string]string {
...
 	if r == nil || k == "" {
-		openlogging.GetLogger().Error("get x_cse_header failed , request is nil or key is empty, please check its")
+		openlogging.GetLogger().Error("get x-cse-header failed , request is nil or key is empty, please check its")
 		return ""
 	}
 	cseContextStr := r.Header.Get(HeaderXCseContent)
 	if cseContextStr == "" {
-		openlogging.GetLogger().Warn("did not has x-cse-header in req")
+		openlogging.GetLogger().Debug("did not has x-cse-header in req")
```
Comment (tianxiaoliang) [inline core/common/common.go]: just delete this log

File: core/common/common.go [hunk 5]
```diff
@@ -220,25 +226,33 @@ func GetXCSEHeader(k string, r *http.Request) string {
 }
 
 // SetXCSEHeader  set value into x-cse-header
-func SetXCSEHeader(k, v string, r *http.Request) {
-	if v == "" || k == "" || r == nil {
-		openlogging.GetLogger().Warn("set x_cse_header into req failed ,because one of key,value and request is empty(nil) or all empty(nil)")
+func SetXCSEHeader(vm map[string]string, r *http.Request) {
```
Comment (tianxiaoliang) [inline core/common/common.go]: SetXCSEHeader --> SetXCSEContext

**Comment by dengguojie** [comment]
> Can one of the admins verify this patch?
I have verified this in provider's first chain handler. 
But still not work ...

**Comment by dengguojie** [comment]
> Can one of the admins verify this patch?

I have verified this in provider's first chain handler.
But still not work ...

**Comment by dengguojie** [comment]
> > Can one of the admins verify this patch?
> 
> I have verified this in some micro-service provider's first chain handler with v1.4.2, which is the latest version.
> But still not work ...

I test with a Consumer of  version v1.3.0 and a Provider of version v1.4.2

**Pull request merged**